app/services/bulk_url_uploader.rb

```diff
@@ -35,7 +36,9 @@ def total_count
     end
 
     def error_messages
-      @errors.keys
+      already_taken = ->(error) { error == URL_ALREADY_TAKEN_MESSAGE }
+      errors = @errors.keys
+      errors.reject(&already_taken) + errors.select(&already_taken)
```

Comment:
To avoid future head-scratching, please add a comment about why the re-ordering is necessary.

Reply:
Comment added.